### ISSUE TITLE
The `"org.apache.airflow.main-image.commit-sha"` image label is empty

### ISSUE THREAD

**Issue by MatrixManAtYrService** [apache/airflow#32291]
### Apache Airflow version

2.3.0 - 2.6.2

### What happened

Regarding the images that are released on dockerhub, I've noticed that the image label which contains the commit is now an empty string.  It was correctly populated in 2.2.4:

```bash
$ docker buildx imagetools inspect docker.io/apache/airflow:2.2.5 --format '{{ json (index .Image.Config) }}' | gron | grep commit
    json.Labels["org.apache.airflow.main-image.commit-sha"] = "7c3513903462511212df742263a49ec1f729ec0e";

$ docker buildx imagetools inspect docker.io/apache/airflow:2.3.0 --format '{{ json (index .Image "linux/amd64").Config }}' | gron | grep commit
    json.Labels["org.apache.airflow.main-image.commit-sha"] = "";
```

### What you think should happen instead

The `airflow.main-image.commit-sha` label should tell me which git ref goes with the image.

### How to reproduce

inspect the released image's labels

### Operating System

any

### Versions of Apache Airflow Providers

n/a

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
cc: @nathadfield -> I have not realised you self-assigned it, so I fixed it in https://github.com/apache/airflow/pull/32949, sorry - but I think it's good we get it fixed before 2.7.0 :). 

You might take a look what it involved though.

**Comment by nathadfield** [comment]
No worries. I've been off for a while so not had time to spend on it.
